src/v/cluster/partition_balancer_planner.cc

```diff
@@ -253,7 +253,9 @@ result<allocation_units> partition_balancer_planner::get_reallocation(
       assignments.group, assignments.id, stable_replicas);
 
     auto reallocation = _partition_allocator.reallocate_partition(
-      std::move(constraints), stable_assigments);
+      std::move(constraints),
+      stable_assigments,
+      model::topic_namespace_view(ntp) == model::kafka_consumer_offsets_nt);
```

Comment:
It seems a bit awkward to have the logic for "is this priority" be outside the allocator: after all, it's a bug if any caller ever uses different logic than other callers, right? So maybe it is better to pass in the ntp and have this logic happen in one place in the allocator, though that also seems ugly in some sense since "why should allocator care about partition names".

This is kind of a recurring theme with the allocator: much of the state is passed in anew at each call site though it generally doesn't make sense to pass in different state, unless something has changed.

At least, we can make an "is_priority" function somewhere and pass in the ntp, and do the `==` once in there...

Reply:
Totally agree. From the first two points I have selected the alloator to be agnostic of specific topic names as having a bigger value.

I'm thinking of replacing the `bool` "priority" flag with an enum that will denote an "allocation domain". That will improve selfdocness of the code and will also make some sense for introduction of more priority allocation domains in future if needed. And also following your suggestion to add a function to determine allocation domain by ntp.

Reply:
Perhaps a domain enum would be a bit cleaner. Would it still have the hierarchical behavior where vanilla partitions allocate themselves taking CGCs into account but not the other way around?

Reply:
Yes it would, the behaviour won't change, only the interface.